src/@types/partials.ts

```diff
@@ -87,3 +87,13 @@ export interface IUsageLimit {
     // eslint-disable-next-line camelcase
     admin_contact?: string;
 }
+
+export enum KnownMembership {
+    Ban = "ban",
+    Invite = "invite",
+    Join = "join",
+    Knock = "knock",
+    Leave = "leave",
+}
+
+export type Membership = KnownMembership | string;
```

Comment:
could you give these things docstrings please.

Reply:
Any idea what `@types/partials` is meant to be about, and how it differs from (say) `@types/event` ?

Also: I slightly wonder whether this should be in the new [types](https://pr4107--js-sdk-docs-previews.netlify.app/modules/types.html) entrypoint rather than adding to the clutter of the main namespace, but then apparently that is meant to be types-only which makes it no god for `enum`s. OTOH, I'm not convinced that defining the new entrypoint to be types-only was a good decision, for this reason.

WDYT?

Reply:
> Any idea what @types/partials is meant to be about, and how it differs from (say) @types/event ?

No. The original contributor put it here, so I left it.

Reply:
> I slightly wonder whether this should be in the new [types](https://pr4107--js-sdk-docs-previews.netlify.app/modules/types.html) entrypoint

I haven't been following this conversation closely, so I don't know and am happy to be guided. Is the types-only decision documented somewhere? Surely an enum is a type? Or does "type" here mean "stuff that disappears entirely at compile time"?

Reply:
An enum is not a type, it maps to an Object literal in JS

Reply:
Docstrings added in d1cc8e9b1

Reply:
Done in 485f69ccb6682713bf78792a2bf761d0c9afaa30